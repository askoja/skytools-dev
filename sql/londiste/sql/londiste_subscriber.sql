
set client_min_messages = 'warning';
\set VERBOSITY 'terse'

--
-- tables
--
create table slavedata (
    id serial primary key,
    data text
);

select current_database();

select * from pgq_set.add_member('branch_set', 'snode', 'dbname=db', false);
select * from pgq_set.add_member('branch_set', 'pnode', 'dbname=db2', false);
select * from pgq_set.create_node('branch_set', 'branch', 'snode', 'londiste_branch', 'pnode', 100, null::text);

select * from londiste.node_add_table('branch_set', 'public.slavedata');
select * from londiste.set_add_table('branch_set', 'public.slavedata');
select * from londiste.node_add_table('branch_set', 'public.slavedata');
select * from londiste.node_get_table_list('branch_set');
select * from londiste.node_remove_table('branch_set', 'public.slavedata');
select * from londiste.node_remove_table('branch_set', 'public.slavedata');
select * from londiste.node_get_table_list('branch_set');

