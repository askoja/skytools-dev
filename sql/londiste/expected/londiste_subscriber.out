set client_min_messages = 'warning';
\set VERBOSITY 'terse'
--
-- tables
--
create table slavedata (
    id serial primary key,
    data text
);
select current_database();
 current_database 
------------------
 regression
(1 row)

select * from pgq_set.add_member('branch_set', 'snode', 'dbname=db', false);
 ret_code | ret_note 
----------+----------
      200 | Ok
(1 row)

select * from pgq_set.add_member('branch_set', 'pnode', 'dbname=db2', false);
 ret_code | ret_note 
----------+----------
      200 | Ok
(1 row)

select * from pgq_set.create_node('branch_set', 'branch', 'snode', 'londiste_branch', 'pnode', 100, null::text);
 ret_code | ret_note 
----------+----------
      200 | Ok
(1 row)

select * from londiste.node_add_table('branch_set', 'public.slavedata');
 ret_code |                   ret_note                    
----------+-----------------------------------------------
      400 | Table not registered in set: public.slavedata
(1 row)

select * from londiste.set_add_table('branch_set', 'public.slavedata');
 ret_code | ret_note 
----------+----------
      200 | OK
(1 row)

select * from londiste.node_add_table('branch_set', 'public.slavedata');
 ret_code |           ret_note            
----------+-------------------------------
      200 | Table added: public.slavedata
(1 row)

select * from londiste.node_get_table_list('branch_set');
    table_name    | merge_state | custom_snapshot | skip_truncate 
------------------+-------------+-----------------+---------------
 public.slavedata |             |                 | 
(1 row)

select * from londiste.node_remove_table('branch_set', 'public.slavedata');
 ret_code |            ret_note             
----------+---------------------------------
      200 | Table removed: public.slavedata
(1 row)

select * from londiste.node_remove_table('branch_set', 'public.slavedata');
 ret_code |          ret_note           
----------+-----------------------------
      400 | Not found: public.slavedata
(1 row)

select * from londiste.node_get_table_list('branch_set');
 table_name | merge_state | custom_snapshot | skip_truncate 
------------+-------------+-----------------+---------------
(0 rows)

